linkerd/app/src/tap.rs

```diff
@@ -49,9 +51,20 @@ impl Config {
             } => {
                 let (listen_addr, listen) = config.bind.bind()?;
 
-                let accept = detect::Accept::new(
-                    tls::DetectTls::new(identity, Default::default()),
-                    tap::AcceptPermittedClients::new(permitted_peer_identities.into(), server),
+                let service =
+                    tap::AcceptPermittedClients::new(permitted_peer_identities.into(), server);
+                let accept = tls::DetectTls::new(
+                    identity,
+                    service_fn(move |meta: tls::accept::Meta| {
+                        let service = service.clone();
+                        future::ok::<_, Never>(service_fn(move |io: io::BoxedIo| {
+                            let fut = service.clone().oneshot((meta.clone(), io));
```

Comment:
Nit: do we need to clone `meta` here? isn't it already being moved into this closure from the outer one?

Reply:
FnOnce vs FnMut

Reply:
oh whoops, yeah. nevermind!